src/core/public/chrome/ui/header/collapsible_nav.tsx

```diff
@@ -69,35 +75,53 @@ function setIsCategoryOpen(id: string, isOpen: boolean, storage: Storage) {
 }
 
 interface Props {
+  appId$: InternalApplicationStart['currentAppId$'];
+  basePath: HttpStart['basePath'];
+  id: string;
   isLocked: boolean;
   isOpen: boolean;
-  navLinks: NavLink[];
-  recentNavLinks: RecentNavLink[];
   homeHref: string;
-  id: string;
+  legacyMode: boolean;
+  navLinks$: Rx.Observable<ChromeNavLink[]>;
+  recentlyAccessed$: Rx.Observable<ChromeRecentlyAccessedHistoryItem[]>;
   storage?: Storage;
   onIsLockedUpdate: OnIsLockedUpdate;
-  onIsOpenUpdate: (isOpen?: boolean) => void;
-  navigateToApp: (appId: string) => void;
+  closeNav: () => void;
+  navigateToApp: InternalApplicationStart['navigateToApp'];
 }
 
 export function CollapsibleNav({
+  basePath,
+  id,
   isLocked,
   isOpen,
-  navLinks,
-  recentNavLinks,
-  onIsLockedUpdate,
-  onIsOpenUpdate,
   homeHref,
-  id,
-  navigateToApp,
+  legacyMode,
   storage = window.localStorage,
+  onIsLockedUpdate,
+  closeNav,
+  navigateToApp,
+  ...observables
 }: Props) {
+  const navLinks = useObservable(observables.navLinks$, []).filter(link => !link.hidden);
+  const recentlyAccessed = useObservable(observables.recentlyAccessed$, []);
+  const appId = useObservable(observables.appId$, '');
   const lockRef = useRef<HTMLButtonElement>(null);
   const groupedNavLinks = groupBy(navLinks, link => link?.category?.id);
   const { undefined: unknowns = [], ...allCategorizedLinks } = groupedNavLinks;
   const categoryDictionary = getAllCategories(allCategorizedLinks);
   const orderedCategories = getOrderedCategories(allCategorizedLinks, categoryDictionary);
+  const readyForEUI = (link: ChromeNavLink, needsIcon: boolean = false) => {
+    return createEuiListItem({
+      link,
+      legacyMode,
+      appId,
+      dataTestSubj: 'collapsibleNavAppLink',
+      navigateToApp,
+      onClick: closeNav,
+      ...(needsIcon && { basePath }),
```

Comment:
Probably minor, but I find it a little misleading that the addition of the icon in `createEuiListItem` is only based on the presence or not of the `basePath` input.